Default Branch Checking Bugfix

rohankh532 opened the pull request:
PR to fix #169

naveensrinivasan reviewed (approved):
Looks good. @azeemshaikh38 👀

naveensrinivasan commented:
Shouldn't this target golang-staging branch?

rohankh532 commented:
> Shouldn't this target golang-staging branch?

Oops, fixed.

azeemshaikh38 commented:
> > Shouldn't this target golang-staging branch?
> 
> Oops, fixed.

This should not be `golang-staging`, merging this to `main` is the right thing to do. Copying my reply to Naveen over slack here: 

> we want to merge these changes to main  and after that to golang-staging . Basically, the only diff between main  and golang-staging  should be that the Dockerfile uses Golang instead of Bash. Everything else should be equivalent

rohankh532 commented:
> > > Shouldn't this target golang-staging branch?
> > 
> > 
> > Oops, fixed.
> 
> This should not be `golang-staging`, merging this to `main` is the right thing to do. Copying my reply to Naveen over slack here:
> 
> > we want to merge these changes to main  and after that to golang-staging . Basically, the only diff between main  and golang-staging  should be that the Dockerfile uses Golang instead of Bash. Everything else should be equivalent

Got it, changed back to main.

azeemshaikh38 commented on Dockerfile:
```diff
@@ -21,23 +21,30 @@
-# Build our image and update the root certs.
-# TODO: use distroless.
-FROM debian:11.3-slim@sha256:f75d8a3ac10acdaa9be6052ea5f28bcfa56015ff02298831994bd3e6d66f7e57
+#v1.17 go
+FROM golang@sha256:bd9823cdad5700fb4abe983854488749421d5b4fc84154c30dae474100468b85 AS base
+WORKDIR /src
+ENV CGO_ENABLED=0
+COPY go.* ./
+RUN go mod download
+COPY . ./
+
+FROM base AS build
+ARG TARGETOS
+ARG TARGETARCH
+RUN CGO_ENABLED=0 make build
```
Dockerfile shouldn't be updated?

rohankh532 replied:
Reverted this

azeemshaikh38 reviewed (changes requested):
LGTM overall. Revert the changes on Dockerfile and ensure that the unit tests are passing.

azeemshaikh38 reviewed (approved):
Thanks!
